cmd/gh/main.go

```diff
@@ -319,6 +318,39 @@ func printError(out io.Writer, err error, cmd *cobra.Command, debug bool) {
 	}
 }
 
+// printNoAuthHelp will print a help message when authentication is required,
+// which is either a welcome message in interactive environments or help on how
+// to set up the non-interactive environment.
+func printNoAuthHelp(out io.Writer, cs *iostreams.ColorScheme) {
+	var helpMessage string
+	if os.Getenv("CI") == "true" {
+		helpMessage = "Either `GH_TOKEN` or `GITHUB_TOKEN` must be set to use `gh`"
+		if os.Getenv("GITHUB_ACTION") != "" {
```

Comment:
Please check for GITHUB_ACTIONS (plural) https://docs.github.com/en/actions/learn-github-actions/environment-variables#default-environment-variables
```suggestion 
		if os.Getenv("GITHUB_ACTIONS") == "true" {
```